dj/construction/dj_query.py

```diff
@@ -0,0 +1,110 @@
+"""
+Functions for making queries directly against DJ
+"""
+
+from typing import Optional, Tuple, cast
+
+from sqlmodel import Session
+
+from dj.construction.build import build_ast_for_database
+from dj.construction.utils import amenable_name, get_dj_node, make_name
+from dj.models.database import Database
+from dj.models.node import NodeType
+from dj.sql.parsing import ast
+from dj.sql.parsing.backends.exceptions import DJParseException
+from dj.sql.parsing.backends.sqloxide import parse
+
+
+async def build_dj_metric_query(  # pylint: disable=R0914,R0912
```

Comment:
I love where this is going! Some early feedback is I think we should split the validation in this function out to a separate function and then call that in a `validate_dj_query` function in `dj/api/helpers.py` and even a `"/sql/validate/"` endpoint.